webapi/src/main/scala/org/knora/webapi/responders/admin/GroupsResponderADM.scala

```diff
@@ -357,58 +352,38 @@ final case class GroupsResponderADMLive(
    * Change group's basic information.
    *
    * @param groupIri             the IRI of the group we want to change.
-   * @param changeGroupRequest   the change request.
-   * @param requestingUser       the user making the request.
+   * @param request              the change request.
    * @param apiRequestID         the unique request ID.
    * @return a [[GroupGetResponseADM]].
    */
-  override def changeGroupBasicInformationRequestADM(
-    groupIri: IRI,
-    changeGroupRequest: GroupUpdatePayloadADM,
-    requestingUser: User,
+  override def updateGroup(
+    groupIri: GroupIri,
+    request: GroupUpdateRequest,
     apiRequestID: UUID
   ): Task[GroupGetResponseADM] = {
+    val task = for {
+      // check if necessary information is present
+      _ <- ZIO
+             .fail(BadRequestException("Group IRI cannot be empty"))
+             .when(groupIri.value.isEmpty)
 
-    /**
-     * The actual change group task run with an IRI lock.
-     */
-    def changeGroupTask(
-      groupIri: IRI,
-      changeGroupRequest: GroupUpdatePayloadADM,
-      requestingUser: User
-    ): Task[GroupGetResponseADM] =
-      for {
-        // check if necessary information is present
-        _ <- ZIO
-               .fail(BadRequestException("Group IRI cannot be empty"))
-               .when(groupIri.isEmpty)
-
-        /* Get the project IRI which also verifies that the group exists. */
-        groupADMOpt <- groupGetADM(groupIri)
-        groupADM <- ZIO
-                      .fromOption(groupADMOpt)
-                      .orElseFail(NotFoundException(s"Group <$groupIri> not found. Aborting update request."))
-
-        /* check if the requesting user is allowed to perform updates */
-        _ <- ZIO
-               .fail(ForbiddenException("Group's information can only be changed by a project or system admin."))
-               .when {
-                 val userPermissions = requestingUser.permissions
-                 !userPermissions.isProjectAdmin(groupADM.project.id) && !userPermissions.isSystemAdmin
-               }
-
-        /* create the update request */
-        groupUpdatePayload = GroupUpdatePayloadADM(
-                               name = changeGroupRequest.name,
-                               descriptions = changeGroupRequest.descriptions,
-                               status = changeGroupRequest.status,
-                               selfjoin = changeGroupRequest.selfjoin
-                             )
-        result <- updateGroupADM(groupIri, groupUpdatePayload)
-      } yield result
-
-    val task = changeGroupTask(groupIri, changeGroupRequest, requestingUser)
-    IriLocker.runWithIriLock(apiRequestID, groupIri, task)
+      /* Get the project IRI which also verifies that the group exists. */
+      groupADMOpt <- groupGetADM(groupIri.value)
+      _ <- ZIO
+             .fromOption(groupADMOpt)
+             .orElseFail(NotFoundException(s"Group <${groupIri.value}> not found. Aborting update request."))
```

Comment:
**suggestion (cosmetic; non-blocking):** I think this could be simplified with something along the lines of
```scala
_ <- ZIO.fail(NotFoundException(s"Group <${groupIri.value}> not found. Aborting update request.")
          .whenZIO(groupGetADM(groupIri.value).map(_.isEmpty))
```
which at least spares you the not-so-nice variable name "groupADMOpt"

Reply:
Thanks for the hint, done in [3c2b7a9](https://github.com/dasch-swiss/dsp-api/pull/3071/commits/3c2b7a9da295c13439240357aaf9edc722de33da)